lib/mix/tasks/compile.thrift.ex

```diff
@@ -27,32 +31,40 @@ defmodule Mix.Tasks.Compile.Thrift do
       source files will be generated. Defaults to `"lib"`.
   """
 
+  @switches [force: :boolean, verbose: :boolean]
+
   @spec run(OptionParser.argv) :: :ok
   def run(args) do
-    {opts, _, _} = OptionParser.parse(args,
-      switches: [force: :boolean, verbose: :boolean])
+    {opts, _, _} = OptionParser.parse(args, switches: @switches)
 
-    config        = Keyword.get(Mix.Project.config, :thrift, [])
-    files         = Keyword.get(config, :files, [])
-    output_path   = Keyword.get(config, :output_path, "lib")
-    parser_opts   = Keyword.take(config, [:include_paths, :namespace])
+    config      = Keyword.get(Mix.Project.config, :thrift, [])
+    files       = Keyword.get(config, :files, [])
+    output_path = Keyword.get(config, :output_path, "lib")
+    parser_opts = Keyword.take(config, [:include_paths, :namespace])
 
-    file_groups =
+    targets =
       files
       |> Enum.map(&parse(&1, parser_opts))
       |> Enum.reject(&is_nil/1)
+      |> extract_targets(output_path, opts[:force])
 
-    stale_groups = Enum.filter(file_groups, fn file_group ->
-      opts[:force] || stale?(file_group, output_path)
-    end)
+    generate(manifest(), targets, output_path, opts)
+  end
 
-    unless Enum.empty?(stale_groups) do
-      File.mkdir_p!(output_path)
-      Mix.Utils.compiling_n(length(stale_groups), :thrift)
-      Enum.each(stale_groups, &generate(&1, output_path, opts))
-    end
+  @doc "Returns the Thrift compiler's manifests."
+  @spec manifests :: [Path.t]
+  def manifests, do: [manifest()]
+  defp manifest, do: Path.join(Mix.Project.manifest_path, @manifest)
+
+  @doc "Cleans up generated files."
+  @spec clean :: :ok | {:error, File.posix}
+  def clean, do: clean(manifest())
+  defp clean(manifest) do
+    Enum.each(read_manifest(manifest), &File.rm/1)
+    File.rm(manifest)
   end
 
+  @spec parse(Path.t, OptionParser.parsed) :: FileGroup.t
```

Comment:
Ironically, I added these when I played with moving some logic into another module, where these functions were public. I continued to find them useful as a way to document the internal arguments.